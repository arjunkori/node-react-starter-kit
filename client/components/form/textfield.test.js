import React from 'react';
import renderer from 'react-test-renderer';

import Textfield from './Textfield';

test('Render textfield without any props', () => {
  const component = renderer.create(
    <Textfield id="textfield-id" />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Render textfield with placeholder', () => {
  const component = renderer.create(
    <Textfield
      id="textfield-id"
      palceholder="Placeholder"
    />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Render disable textfield', () => {
  const component = renderer.create(
    <Textfield
      id="textfield-id"
      isDisabled
    />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
