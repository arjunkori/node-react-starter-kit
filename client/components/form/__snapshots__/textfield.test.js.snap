// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render disable textfield 1`] = `
<input
  autoComplete="off"
  className="textfield disabled"
  defaultValue=""
  disabled={true}
  id="textfield-id"
  name={null}
  onChange={null}
  placeholder=""
  tabIndex={null}
  type="text"
/>
`;

exports[`Render textfield with placeholder 1`] = `
<input
  autoComplete="off"
  className="textfield"
  defaultValue=""
  disabled={false}
  id="textfield-id"
  name={null}
  onChange={null}
  placeholder=""
  tabIndex={null}
  type="text"
/>
`;

exports[`Render textfield without any props 1`] = `
<input
  autoComplete="off"
  className="textfield"
  defaultValue=""
  disabled={false}
  id="textfield-id"
  name={null}
  onChange={null}
  placeholder=""
  tabIndex={null}
  type="text"
/>
`;
